src/main/java/org/broadinstitute/hellbender/engine/ReadWindow.java

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.iterators.ReadFilteringIterator;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A class to represent a window of reads data, optionally expanded by a configurable amount of padded data.
+ *
+ * The reads are lazily loaded by default (when accessing the reads via {@link #iterator}. Loading all the
+ * reads in the window at once is possible via {@link #loadAllReads}.
+ *
+ * The reads returned will overlap the expanded padded window. It's possible to query whether they are within
+ * the main part of the window via {@link #isContainedWithinWindow} and {@link #startsWithinWindow}.
+ *
+ * The reads in the window can be filtered via {@link #setReadFilter} (no filtering is performed by default).
+ */
+public final class ReadWindow implements Iterable<GATKRead>, Locatable {
```

Comment:
They are deliberately separate abstractions, since they serve different purposes. `ReadWindow` is supposed to be the unit of eventual parallelism for all `ReadWindowWalkers`, while `AssemblyRegion` is a work unit for the HC specifically. We may eventually settle on a strategy involving creating multiple `AssemblyRegions` per `ReadWindow`, instead of the current 1:1 correspondence.